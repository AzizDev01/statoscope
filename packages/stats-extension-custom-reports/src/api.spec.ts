import APIFactory from './api';
import Generator from './generator';

test('should work', () => {
  const generator = new Generator();

  generator.handleReport({
    context: 'context',
    data: 'data',
    id: 'foo',
    name: 'name',
    view: ['some-ui'],
  });

  generator.handleReport({
    context: 'context2',
    data: 'data2',
    id: 'foo2',
    name: 'name2',
    view: ['some-ui2'],
  });

  generator.handleReport({
    context: 'context3',
    data: 'data3',
    id: 'foo3',
    name: 'name3',
    view: ['some-ui3'],
    relations: [{ type: 'module', id: 'foo' }],
  });

  generator.handleReport({
    context: 'context4',
    data: 'data4',
    id: 'foo4',
    name: 'name4',
    view: ['some-ui4'],
    relations: [{ type: 'module', id: 'foo' }],
  });

  generator.handleReport({
    context: 'context5',
    data: 'data5',
    id: 'foo5',
    name: 'name5',
    view: ['some-ui5'],
    relations: [{ type: 'module', id: 'bar' }],
  });

  generator.handleReport({
    context: 'context6',
    data: 'data6',
    id: 'foo6',
    name: 'name6',
    view: ['some-ui6'],
    relations: [{ type: 'chunk', id: 'foo' }],
  });

  generator.handleReport({
    context: 'context7',
    data: 'data7',
    id: 'foo7',
    name: 'name7',
    view: ['some-ui7'],
    compilation: 'foo-compilation',
    relations: [{ type: 'chunk', id: 'foo' }],
  });

  const data = generator.get();
  const api = APIFactory(data);

  expect(api.getReports()).toMatchSnapshot();
  expect(api.getReports(null, 'package')).toEqual([]);
  expect(api.getReports(null, 'module')).toMatchSnapshot();
  expect(api.getReports(null, 'module', 'baz')).toEqual([]);
  expect(api.getReports(null, 'module', 'foo')).toMatchSnapshot();
  expect(api.getReports(null, 'module', 'bar')).toMatchSnapshot();
  expect(api.getReports(null, 'chunk')).toMatchSnapshot();
  expect(api.getReports(null, 'chunk', 'foo')).toMatchSnapshot();
  expect(api.getReports('foo-compilation')).toMatchSnapshot();
  expect(api.getReports('foo-compilation', 'chunk')).toMatchSnapshot();
  expect(api.getReports('foo-compilation', 'module')).toEqual([]);
  expect(api.getReports('foo-compilation', 'chunk', 'foo')).toMatchSnapshot();
  expect(api.getReports('foo-compilation', 'module', 'foo')).toEqual([]);
  expect(api.getById('foo')).toMatchSnapshot();
  expect(api.getById('foo2')).toMatchSnapshot();
  expect(api.getById('foo777')).toBeNull();
});
