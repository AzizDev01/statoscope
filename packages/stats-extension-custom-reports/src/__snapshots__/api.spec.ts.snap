// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Array [
  Object {
    "context": "context",
    "data": "data",
    "id": "foo",
    "name": "name",
    "view": Array [
      "some-ui",
    ],
  },
  Object {
    "context": "context2",
    "data": "data2",
    "id": "foo2",
    "name": "name2",
    "view": Array [
      "some-ui2",
    ],
  },
  Object {
    "context": "context3",
    "data": "data3",
    "id": "foo3",
    "name": "name3",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui3",
    ],
  },
  Object {
    "context": "context4",
    "data": "data4",
    "id": "foo4",
    "name": "name4",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui4",
    ],
  },
  Object {
    "context": "context5",
    "data": "data5",
    "id": "foo5",
    "name": "name5",
    "relations": Array [
      Object {
        "id": "bar",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui5",
    ],
  },
  Object {
    "context": "context6",
    "data": "data6",
    "id": "foo6",
    "name": "name6",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui6",
    ],
  },
]
`;

exports[`should work 2`] = `
Array [
  Object {
    "context": "context3",
    "data": "data3",
    "id": "foo3",
    "name": "name3",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui3",
    ],
  },
  Object {
    "context": "context4",
    "data": "data4",
    "id": "foo4",
    "name": "name4",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui4",
    ],
  },
  Object {
    "context": "context5",
    "data": "data5",
    "id": "foo5",
    "name": "name5",
    "relations": Array [
      Object {
        "id": "bar",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui5",
    ],
  },
]
`;

exports[`should work 3`] = `
Array [
  Object {
    "context": "context3",
    "data": "data3",
    "id": "foo3",
    "name": "name3",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui3",
    ],
  },
  Object {
    "context": "context4",
    "data": "data4",
    "id": "foo4",
    "name": "name4",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui4",
    ],
  },
]
`;

exports[`should work 4`] = `
Array [
  Object {
    "context": "context5",
    "data": "data5",
    "id": "foo5",
    "name": "name5",
    "relations": Array [
      Object {
        "id": "bar",
        "type": "module",
      },
    ],
    "view": Array [
      "some-ui5",
    ],
  },
]
`;

exports[`should work 5`] = `
Array [
  Object {
    "context": "context6",
    "data": "data6",
    "id": "foo6",
    "name": "name6",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui6",
    ],
  },
]
`;

exports[`should work 6`] = `
Array [
  Object {
    "context": "context6",
    "data": "data6",
    "id": "foo6",
    "name": "name6",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui6",
    ],
  },
]
`;

exports[`should work 7`] = `
Array [
  Object {
    "compilation": "foo-compilation",
    "context": "context7",
    "data": "data7",
    "id": "foo7",
    "name": "name7",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui7",
    ],
  },
]
`;

exports[`should work 8`] = `
Array [
  Object {
    "compilation": "foo-compilation",
    "context": "context7",
    "data": "data7",
    "id": "foo7",
    "name": "name7",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui7",
    ],
  },
]
`;

exports[`should work 9`] = `
Array [
  Object {
    "compilation": "foo-compilation",
    "context": "context7",
    "data": "data7",
    "id": "foo7",
    "name": "name7",
    "relations": Array [
      Object {
        "id": "foo",
        "type": "chunk",
      },
    ],
    "view": Array [
      "some-ui7",
    ],
  },
]
`;

exports[`should work 10`] = `
Object {
  "context": "context",
  "data": "data",
  "id": "foo",
  "name": "name",
  "view": Array [
    "some-ui",
  ],
}
`;

exports[`should work 11`] = `
Object {
  "context": "context2",
  "data": "data2",
  "id": "foo2",
  "name": "name2",
  "view": Array [
    "some-ui2",
  ],
}
`;
